extension/src/time_series/pipeline/fill_to.rs

```diff
@@ -0,0 +1,246 @@
+                let left_wt = 1. - (target_ts - lhs.ts) as f64 / interval;
+                let right_wt = 1. - (rhs.ts - target_ts) as f64 / interval;
+                TSPoint{ts: target_ts, val: lhs.val * left_wt + rhs.val * right_wt}
+            },
+            FillToMethod::Nearest =>
+                if rhs.ts - target_ts >= target_ts - lhs.ts {
+                    TSPoint{ts: target_ts, val: lhs.val}
+                } else {
+                    TSPoint{ts: target_ts, val: rhs.val}
+                },
+        }
+    }
+}
+
+// TODO is (immutable, parallel_safe) correct?
+#[pg_extern(
+    immutable,
+    parallel_safe,
+    name="fill_to",
+    schema="toolkit_experimental"
+)]
+pub fn fillto_pipeline_element<'e> (
+    fill_method: String,
+    interval: Interval,
+) -> toolkit_experimental::UnstableTimevectorPipeline<'e> {
+    unsafe {
+        let interval = interval as *const pg_sys::Interval;
+        // TODO: store the postgres interval object and use postgres timestamp/interval functions
+        let interval = ((*interval).month as i64 * 30 + (*interval).day as i64) * 24 * 60 * 60 * 1000000 + (*interval).time;
+
+        let fill_method = match fill_method.to_lowercase().as_str() {
+            "locf" => FillToMethod::LOCF,
+            "interpolate" => FillToMethod::Interpolate,
+            "linear" => FillToMethod::Interpolate,
+            "nearest" => FillToMethod::Nearest,
+            _ => panic!("Invalid fill method")
+        };
+
+        Element::FillTo {
+            interval,
+            fill_method
+        }.flatten()
+    }
+}
+
+pub fn fill_to<'s>(
+    series: &toolkit_experimental::Timevector<'s>,
+    element: &toolkit_experimental::Element
+) -> toolkit_experimental::Timevector<'s> {
+    let (interval, method) = match element {
+        Element::FillTo{interval, fill_method} => (*interval, fill_method),
+        _ => unreachable!()
+    };
+
+    match series.series {
+        SeriesType::ExplicitSeries{..} => panic!("Timeseries must be sorted prior to passing to fill_to"),
+        SeriesType::NormalSeries{step_interval, ..} => if step_interval <= interval { return series.clone(); },
+        _ => ()
+    }
+
+    let mut result: Option<Vec<TSPoint>> = None;
+    let mut it = series.iter().peekable();
+    let mut current = it.next();
+
+    while current.is_some() {
+        let lhs = current.unwrap();
+        let rhs = it.peek();
+
+        if rhs.is_some() && rhs.unwrap().ts - lhs.ts > interval {
+            let rhs = rhs.unwrap();
+            if result.is_none() {
+                result = Some(series.iter().collect());
+            }
+            let result = result.as_mut().unwrap();
+            let mut target = lhs.ts + interval;
+            while target < rhs.ts {
+                result.push(method.fill_point(&lhs, &rhs, target));
+                target += interval;
+            }
+        }
+
+        current = it.next();
+    }
```

Comment:
Just for fun, if we were only working on slices we could do
```suggestion
    let mut result = vec![];
    let mut windows = series.points.windows(2);

    for window in windows {
        if window[1].ts - window[0].ts > interval {
            let result = result.as_mut().unwrap();
            let mut target = window[0].ts + interval;
            while target < window[1].ts {
                result.push(method.fill_point(&window[0], &window[1], target));
                target += interval;
            }
        }
    }
```